scripts/build.js

```diff
@@ -11,20 +11,30 @@ const scriptPkg = require(`${scriptVersion}/package.json`);
 // https://github.com/facebook/create-react-app/releases/tag/v2.1.2
 const isWebpackFactory = semver.gte(scriptPkg && scriptPkg.version, '2.1.2');
 
-const webpackConfigPath = `${scriptVersion}/config/webpack.config${!isWebpackFactory ? '.prod' : ''}`;
-const webpackConfig = require(webpackConfigPath);
+let webpackConfigPath;
+let webpackConfig;
+
+if (isWebpackFactory) {
+  webpackConfigPath = `${scriptVersion}/config/webpack.config`;
+  webpackConfig = require(webpackConfigPath)('production');
```

Comment:
Seems like this line is the main difference between this PR and the existing code:

```diff
- webpackConfig = require(webpackConfigPath);
+ webpackConfig = require(webpackConfigPath)('production');
```

Seems like this change introduces a bug because `webpackConfig` is invoked on [LOC#27](https://github.com/timarney/react-app-rewired/pull/355/files#diff-ea1c4abd09127cfb80f898b2bf12b0a4R27):

```js
env => overrides.webpack(webpackConfig(env), env)
```

Seems like LOC 27 should error because `require(webpackConfigPath)('production')` returns a webpack configuration object, not a function.